src/app/panels/text-panel/text-panel.component.ts

```diff
@@ -35,6 +35,15 @@ export class TextPanelComponent implements OnInit, OnDestroy {
     filter(e => !!e),
     distinctUntilChanged(),
   );
+  public msDesc$ = this.evtModelService.msDesc$;
+
+  // tslint:disable-next-line: variable-name
+  private _msDescID: string;
+  @Input() set msDescID(p: string) {
+     this._msDescID = p;
+   }
+
+  get msDescID() { return this._msDescID; }
```

Comment:
You don't need lines 38-46 here, because you manage them in msdesc-selector component